docs: Add comprehensive AuthBridge documentation (#1476)
kagenti/kagenti: docs/authbridge/README.md

# AuthBridge

AuthBridge provides platform primitives to secure AI agents by managing agent
identity, authentication, and authorization invisibly — and enforces network
guardrails that scope access to minimum necessary permissions with full
auditability.

**Why it matters:** AI agents make autonomous, non-deterministic decisions about
which tools or services to call, making it unsafe to trust them with secrets or
self-authorization. AuthBridge shifts credential management to the platform layer
where it is enforced (not advisory), surviving prompt injection and agent compromise
while preserving the full identity delegation chain from end user through
orchestrator to tool.

## What AuthBridge Does

> - **Transparent credential injection** — agents never see or manage tokens; AuthBridge

Reviewer: ```suggestion
- **Transparent token injection** — agents never see or manage tokens used in outgoing calls; AuthBridge
```

>   intercepts outbound requests and attaches audience-scoped credentials automatically

Reviewer: ```suggestion
  intercepts outbound requests and attaches audience-scoped tokens automatically
```

> - **Token exchange** — converts workload identity (SPIFFE JWT-SVID or K8s service account

Reviewer: ```suggestion
as an example, AuthBridge may use the agent workload identity (SPIFFE JWT-SVID) and the K8s service account
```

>   token) into short-lived, audience-specific OAuth tokens via RFC 8693

Reviewer: ```suggestion
  token) to obtain a short-lived, audience-specific OAuth tokens via token exchange (RFC 8693)
```

> - **Tool access control** — restricts which external services (MCP tools, APIs, LLMs)

Reviewer: ```suggestion
- **Tool access control** — restricts which external services (MCP tools, Other Agents, APIs, LLMs)
```

  each agent can reach, based on host allowlists and protocol-aware policies
- **Inbound validation** — verifies JWT tokens on incoming requests, ensuring only
  authorized callers can invoke an agent
- **Delegation chains** — preserves the identity chain (user → orchestrator → agent → tool)
  so every hop is authenticated and authorized
- **No privileged mode required** — the default deployment uses a single unprivileged
  sidecar container with standard HTTP_PROXY routing

## How It Works

```
┌─────────────────────────────────────────────────────────────────┐
│  Agent Pod                                                       │
│                                                                  │
│  ┌─────────────┐    HTTP_PROXY     ┌──────────────────────┐     │
│  │   Agent     │ ───────────────── │  AuthBridge Proxy    │     │
│  │ (any        │                   │                      │     │
│  │  framework) │ ◄──── reverse ─── │  - JWT validation    │     │

> │  │             │       proxy       │  - Token exchange    │     │

Reviewer: ```suggestion
│  │             │       proxy       │  - Token injection    │     │
```

│  └─────────────┘                   │  - Access control    │     │
│                                    └──────────┬───────────┘     │
└───────────────────────────────────────────────┼─────────────────┘
                                                │
                    ┌───────────────────────────┼──────────────────┐
                    │                           │                   │
                    ▼                           ▼                   ▼
            ┌──────────────┐         ┌──────────────┐    ┌──────────────┐
            │  Keycloak    │         │  MCP Tools   │    │  LLM APIs    │
            │  (token      │         │  (weather,   │    │  (OpenAI,    │
            │   exchange)  │         │   github...) │    │   Anthropic) │
            └──────────────┘         └──────────────┘    └──────────────┘
```

> 

Reviewer: Show also approaches to other agents
Add "..." to identify that these are examples and there is more.

**Inbound flow:** A request arrives at the agent → AuthBridge validates the caller's
JWT → if valid, forwards to the agent container.

> **Outbound flow:** The agent makes an HTTP call to a tool or LLM → AuthBridge intercepts

Reviewer: ```suggestion
**Outbound flow:** The agent makes an HTTP call to a tool, other agent or LLM → AuthBridge intercepts
```

> via HTTP_PROXY → exchanges the agent's workload identity for an audience-scoped token →

Reviewer: ```suggestion
via HTTP_PROXY → ensures agent is allowed to make the call -> injects an audience-scoped token →
```

> attaches the token and forwards the request.

Reviewer: ```suggestion
forwards the request to the destination.
```

## For End Users

When you interact with an agent, AuthBridge is working behind the scenes to protect you:

> - The agent can only access tools and data it's explicitly authorized for — it cannot

Reviewer: ```suggestion
- The agent can only access tools, agnets and data it's explicitly authorized for — it cannot
```

  reach arbitrary services even if compromised or hallucinating

> - Your identity is carried through the delegation chain so access decisions reflect

Reviewer: ```suggestion
- The calling user identity is carried through the delegation chain so access decisions may reflect
```

>   *your* permissions, not a shared service account

Reviewer: ```suggestion
  the calling user permissions, as well as supporting a shared service account when desired
```

> - Every tool call the agent makes is logged and auditable

Reviewer: ```suggestion
- Every tool call and agent call the agent makes is logged and auditable
```

- The platform enforces these guarantees — the agent cannot bypass them

Learn more: [Security Model](security-model.md)

## For Agent Developers

> You don't need to do anything. AuthBridge handles auth transparently:

Reviewer: ```suggestion
You only need to pass the received token as is to any outbound call you are making. AuthBridge verify the inbound token before you receive it and handles auth based on your inbound token when you call others:
```

- No SDKs to import, no auth code to write

> - Your agent makes normal HTTP calls; credentials are injected automatically

Reviewer: ```suggestion
- Your agent makes normal HTTP calls; adequante credentials are injected automatically
```
